node/src/components/small_network.rs

```diff
@@ -851,6 +886,53 @@ where
         ret
     }
 
+    /// Updates the bootstrapping state of the node according to available information.
+    fn update_bootstrapping_state(&mut self) {
```

Comment:
What about adding a unit test to prove that the calculations, filter_maps, and intersections as a whole work as expected?

I'd suggest introducing a function taking the following parameters:
- self.cfg.bootstrap_threshold
- self.connection_symmetries
- self.cfg.known_addresses
- possibly more (compiler will tell)

Then in `update_bootstrapping_state` you can return early on `self.bootstrap_completed` (as you do) or invoke the proposed function.

This will allow adding unit test with different inputs exercising various edge cases.

Also, to have a single source of truth, I'd suggest adding a check for timeout threshold also here, by having yet another testable `if` instead of an async-based `set_timeout` mechanism.